Gen binds

lum1n0us opened the pull request:
left three things behind
```
wasm_engine_t *
wasm_engine_new_with_args(mem_alloc_type_t type, const MemAllocOption *opts);

typedef enum {
    Alloc_With_Pool = 0,
    Alloc_With_Allocator,
    Alloc_With_System_Allocator,
} mem_alloc_type_t;

typedef union MemAllocOption {
    struct {
        void *heap_buf;
        uint32_t heap_size;
    } pool;
    struct {
        void *malloc_func;
        void *realloc_func;
        void *free_func;
    } allocator;
} MemAllocOption;
```

lum1n0us committed:
Parse Struct and Typedef information

lum1n0us committed:
create a script to genrate bindings

lum1n0us commented:
@BenSunWhiteBoard there is a draft version of `binggen.py`, and you can use it to create all wrappers, and there is a simple introduction to the preparation, and please let me know when you have any doubts, and good luck.

lum1n0us committed:
add an additional

contains wrappers can not generated by bindgen

lum1n0us committed:
upload generated bindings

BenSunWhiteBoard commented:
Actually when I tried to run it myself, I find out an error message like:
```
......
  File "utils/bindgen.py", line 134, in visit_Struct
    info[decl.name] = self.get_type_name(decl.type)
  File "utils/bindgen.py", line 52, in get_type_name
    return self.get_type_name(type.type)
  File "utils/bindgen.py", line 68, in get_type_name
    raise Exception(f"a new type should be in type_map: {type}")
Exception: a new type should be in type_map: long
```
for long and errno_t, after I added
```
            "errno_t": "c_int",
            "long": "c_long",
```
to type_map, it's all fine.

I guess that's because I am using macOS and size of certain type is a little different?

lum1n0us committed:
choose not to generate del function

lum1n0us committed:
upload binding.py

lum1n0us committed:
Add more helper in ffi

- since only wasm_limit_t and wasm_XXX_vec have their strcut definitions,
  We don't need to generate their `__repr__`, and instead of it, we
  implement them in ffi
- Put helpers in ffi.
- rename with prologue

lum1n0us committed:
add a method to return a NULL pointer

lum1n0us committed:
Add helpers for wasm_externtype_t and wasm_importtype_t

lum1n0us committed:
add helpers about wasm_exporttype_t

lum1n0us committed:
Add helpers for wasm_val_t, wasm_trap_t, wasm_module_t

Generate __repr__ for vectors

lum1n0us committed:
add helpers for wasm_func_t

lum1n0us committed:
add a new c function pointer for finalizer and helpers

lum1n0us committed:
add helpers for the rest

lum1n0us committed:
fix some pylint warnings

lum1n0us committed:
enhancement about wasm_frame_t

lum1n0us committed:
remove unused files

lum1n0us committed:
update document
